pkg/inspector/image-inspector.go

```diff
@@ -126,131 +136,70 @@ func (i *defaultImageInspector) Inspect() error {
+			results, reportObj, err = scanner.Scan(ctx, i.opts.DstPath, &i.meta.Image, filterFn)
+			if err != nil {
+				i.meta.OpenSCAP.SetError(err)
+				log.Printf("DEBUG: Unable to scan image %q with OpenSCAP: %v", i.opts.Image, err)
+			} else {
+				i.meta.OpenSCAP.Status = iiapi.StatusSuccess
+				report := reportObj.(openscap.OpenSCAPReport)
+				scanReport = report.ArfBytes
+				htmlScanReport = report.HTMLBytes
+				scanResults.Results = append(scanResults.Results, results...)
 			}
 
-			for _, prefix := range excludePrefixes {
-				if strings.HasPrefix(path, prefix) {
-					return false
-				}
+		case "clamav":
+			scanner, err = clamav.NewScanner(i.opts.ClamSocket)
+			if err != nil {
+				return fmt.Errorf("failed to initialize clamav scanner: %v", err)
+			}
+			results, _, err := scanner.Scan(ctx, i.opts.DstPath, &i.meta.Image, filterFn)
+			if err != nil {
+				log.Printf("DEBUG: Unable to scan image %q with ClamAV: %v", i.opts.Image, err)
+				return err
 			}
-
-			return true
-		}
-	}
-
-	switch i.opts.ScanType {
-	case "openscap":
-		if i.opts.ScanResultsDir, err = createOutputDir(i.opts.ScanResultsDir, "image-inspector-scan-results-"); err != nil {
-			return err
-		}
-		var (
-			results   []iiapi.Result
-			reportObj interface{}
-		)
-		scanner = openscap.NewDefaultScanner(OSCAP_CVE_DIR, i.opts.ScanResultsDir, i.opts.CVEUrlPath, i.opts.OpenScapHTML)
-		results, reportObj, err = scanner.Scan(ctx, i.opts.DstPath, &i.meta.Image, filterFn)
-		if err != nil {
-			i.meta.OpenSCAP.SetError(err)
-			log.Printf("DEBUG: Unable to scan image %q with OpenSCAP: %v", i.opts.Image, err)
-		} else {
-			i.meta.OpenSCAP.Status = iiapi.StatusSuccess
-			report := reportObj.(openscap.OpenSCAPReport)
-			scanReport = report.ArfBytes
-			htmlScanReport = report.HTMLBytes
 			scanResults.Results = append(scanResults.Results, results...)
-		}
 
-	case "clamav":
-		scanner, err = clamav.NewScanner(i.opts.ClamSocket)
-		if err != nil {
-			return fmt.Errorf("failed to initialize clamav scanner: %v", err)
-		}
-		results, _, err := scanner.Scan(ctx, i.opts.DstPath, &i.meta.Image, filterFn)
-		if err != nil {
-			log.Printf("DEBUG: Unable to scan image %q with ClamAV: %v", i.opts.Image, err)
-			return err
+		default:
+			return fmt.Errorf("unsupported scan type: %s", i.opts.ScanType)
 		}
-		scanResults.Results = append(scanResults.Results, results...)
 
-	default:
-		return fmt.Errorf("unsupported scan type: %s", i.opts.ScanType)
-	}
-
-	if len(i.opts.PostResultURL) > 0 {
-		if err := i.postResults(scanResults); err != nil {
-			log.Printf("Error posting results: %v", err)
-			return nil
+		if len(i.opts.PostResultURL) > 0 {
+			if err := i.postResults(scanResults); err != nil {
+				log.Printf("Error posting results: %v", err)
+				return nil
+			}
 		}
 	}
 
 	if i.imageServer != nil {
-		return i.imageServer.ServeImage(&i.meta, i.opts.DstPath, scanResults, scanReport, htmlScanReport)
+		if i.meta.ImageAcquireSuccess {
```

Comment:
@enoodle it's not easy to understand why you need this check now. In general I would try to encapsulate any further logic in `ServeImage`.

Reply:
Will move to the `ServeImage` function.